theme/ckeditor5-table/tableediting.css

```diff
@@ -3,23 +3,27 @@
  * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
  */
 
-:root {
-	--ck-color-table-focused-cell-background: hsl(208, 90%, 98%);
-}
-
 .ck-widget.table {
-	& td,
-	& th {
-		&.ck-editor__nested-editable.ck-editor__nested-editable_focused {
-			/* A very slight background to highlight the focused cell */
-			background: var(--ck-color-table-focused-cell-background);
+	& .ck-editor__nested-editable.ck-editor__nested-editable_focused,
+	& .ck-editor__nested-editable:focus {
+		background: inherit;
 
-			/* Fixes the problem where surrounding cells cover the focused cell's border.
-			It does not fix the problem in all places but the UX is improved.
-			See https://github.com/ckeditor/ckeditor5-table/issues/29. */
-			border-style: none;
-			outline: 1px solid var(--ck-color-focus-border);
-			outline-offset: -1px; /* progressive enhancement - no IE support */
+		/* Fixes the problem where surrounding cells cover the focused cell's border.
+		It does not fix the problem in all places but the UX is improved.
+		See https://github.com/ckeditor/ckeditor5-table/issues/29. */
+		border-style: none;
+		outline: 1px solid var(--ck-color-focus-border);
+		outline-offset: -1px; /* progressive enhancement - no IE support */
+	}
+
+	& th {
+		&.ck-editor__nested-editable.ck-editor__nested-editable_focused,
+		&.ck-editor__nested-editable:focus {
+			/* We can't just set `inherit` here as well, because we use `hsla` as an "overlay"
+			to darken the background based on the set table background.
+			This is also a workaround for overriding the styling that comes from the `widget`.
+			See related issue https://github.com/ckeditor/ckeditor5/issues/6228. */
+			background: var(--ck-color-table-focused-th-background);
```

Comment:
As commented in ckeditor/ckeditor5-table#269 – this sounds really wrong. I don't understand why we need to refactor here anything.

Reply:
The `:focus` selector prevents the FOUC because the `_focused` class is applied with a delay by the engine. It looks nasty and this is OK.

But I agree the overall fix should be simpler, like this

```diff
diff --git a/theme/ckeditor5-table/tableediting.css b/theme/ckeditor5-table/tableediting.css
index 01966ab..9463ad9 100644
--- a/theme/ckeditor5-table/tableediting.css
+++ b/theme/ckeditor5-table/tableediting.css
@@ -4,13 +4,14 @@
  */

 :root {
-         --ck-color-table-focused-cell-background: hsl(208, 90%, 98%);
+         --ck-color-table-focused-cell-background: hsla(208, 90%, 80%, .3);
 }

 .ck-widget.table {
          & td,
          & th {
-                  &.ck-editor__nested-editable.ck-editor__nested-editable_focused {
+                  &.ck-editor__nested-editable.ck-editor__nested-editable_focused,
+                  &.ck-editor__nested-editable:focus {
                            /* A very slight background to highlight the focused cell */
                            background: var(--ck-color-table-focused-cell-background);
```

Reply:
![2020-03-11 11 06 31](https://user-images.githubusercontent.com/1099479/76405474-77830a00-6388-11ea-9995-4f93041c0cd4.gif)

Reply:
@oleq I like the `:focus` color in your example, but when we have a cell with a color set by JS (inline styles) `:focus ` set by the CSS is always overridden and we don't see any changes... So do we need to replace the background anyway? I'd just reset it. We have other indicators that the cell is focused.